test/__init__.py

```diff
@@ -59,26 +59,25 @@ def run_role_meta_main(self, meta_main_text):
 
 
 def run_ansible_lint(
+        *argv: str,
         cwd: os.PathLike = None,
-        role_path: os.PathLike = None,
         bin: str = None,
-        env: Dict[str, str] = None):
+        env: Dict[str, str] = None) -> Tuple[str, str]:
     """Run ansible-lint on a given path and returns its output."""
     if bin:
-        command = [bin, "-v"]
+        command = [bin]
     else:
-        command = [sys.executable, "-m", "ansiblelint", "-v"]
-    if role_path:
-        command.append(str(role_path))
+        command = [sys.executable, "-m", "ansiblelint"]
+    if argv:
+        command.extend(argv)
 
-    result, err = subprocess.Popen(
+    return subprocess.Popen(
         command,
         cwd=cwd,
         stdin=subprocess.PIPE,
         stdout=subprocess.PIPE,
         stderr=subprocess.PIPE,
         shell=False,  # needed when command is a list
         env=env,
+        universal_newlines=True
     ).communicate()
```

Comment:
Since you're already refactoring this...
The docs recommend using run instead of Popen starting Python 3.5: https://docs.python.org/3/library/subprocess.html#using-the-subprocess-module.
```suggestion
    proc_result = subprocess.run(
        command,
        cwd=cwd,
        stdin=subprocess.PIPE,
        stdout=subprocess.PIPE,
        stderr=subprocess.PIPE,
        shell=False,  # needed when command is a list
        env=env,
        universal_newlines=True
    )
    return proc_result.stdout, proc_result.stderr
```
(Although, I'd probably return the whole `subprocess.CompletedProcess` instance)